include/swift/AST/Decl.h

```diff
@@ -5805,6 +5805,10 @@ class AbstractFunctionDecl : public GenericContext, public ValueDecl {
   struct {
     unsigned NeedsNewVTableEntryComputed : 1;
     unsigned NeedsNewVTableEntry : 1;
+
+    /// Whether this is an Objective-C declaration and has the objc_direct
+    /// attribute.
+    unsigned IsObjCDirect : 1;
```

Comment:
I don't think we need to leak details of the calling convention all the way up into the surface language's AST. We can pull this information out of the corresponding clang node at [SIL/IR]Gen time.